server/command_test.go

```diff
@@ -0,0 +1,160 @@
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/plugin"
+	"github.com/mattermost/mattermost-server/v6/plugin/plugintest"
+	"github.com/pkg/errors"
+
+	"github.com/stretchr/testify/mock"
+)
+
+func TestExecuteUnlinkCommand(t *testing.T) {
+	p := newTestPlugin()
+	mockAPI := &plugintest.API{}
+
+	for _, testCase := range []struct {
+		description string
+		args        *model.CommandArgs
+		setupAPI    func(*plugintest.API)
+		setupStore  func(*mockStore.Store)
+	}{
+		{
+			description: "Successfully executed unlinked command",
+			args: &model.CommandArgs{
+				UserId:    testutils.GetUserID(),
+				ChannelId: testutils.GetChannelID(),
+			},
+			setupAPI: func(api *plugintest.API) {
+				api.On("GetChannel", testutils.GetChannelID()).Return(&model.Channel{
+					Id:   testutils.GetChannelID(),
+					Type: model.ChannelTypeOpen,
+				}, nil).Times(1)
+				api.On("HasPermissionToChannel", testutils.GetUserID(), testutils.GetChannelID(), model.PermissionManagePublicChannelProperties).Return(true).Times(1)
+			},
+			setupStore: func(s *mockStore.Store) {
+				s.On("DeleteLinkByChannelID", testutils.GetChannelID()).Return(nil).Times(1)
+			},
+		},
+		{
+			description: "Unable to delete link.",
+			args: &model.CommandArgs{
+				UserId:    testutils.GetUserID(),
+				ChannelId: "Mock-ChannelID",
+			},
+			setupAPI: func(api *plugintest.API) {
+				api.On("GetChannel", "Mock-ChannelID").Return(&model.Channel{
+					Id:   "Mock-ChannelID",
+					Type: model.ChannelTypeOpen,
+				}, nil).Times(1)
+				api.On("HasPermissionToChannel", testutils.GetUserID(), "Mock-ChannelID", model.PermissionManagePublicChannelProperties).Return(true).Times(1)
+			},
+			setupStore: func(s *mockStore.Store) {
+				s.On("DeleteLinkByChannelID", "Mock-ChannelID").Return(errors.New("Error while deleting a link")).Times(1)
+			},
+		},
+		{
+			description: "Unable to get the current channel",
+			args:        &model.CommandArgs{},
+			setupAPI: func(api *plugintest.API) {
+				api.On("GetChannel", "").Return(nil, testutils.GetInternalServerAppError("Error while getting the current channel.")).Times(1)
+			},
+			setupStore: func(s *mockStore.Store) {},
+		},
+		{
+			description: "Unable to unlink channel as user is not a channel admin.",
+			args: &model.CommandArgs{
+				ChannelId: testutils.GetChannelID(),
+			},
+			setupAPI: func(api *plugintest.API) {
+				api.On("GetChannel", testutils.GetChannelID()).Return(&model.Channel{
+					Id:   testutils.GetChannelID(),
+					Type: model.ChannelTypeOpen,
+				}, nil).Times(1)
+				api.On("HasPermissionToChannel", "", testutils.GetChannelID(), model.PermissionManagePublicChannelProperties).Return(false).Times(1)
+			},
+			setupStore: func(s *mockStore.Store) {},
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			mockAPI.On("SendEphemeralPost", mock.Anything, mock.Anything).Return(testutils.GetPost())
```

Comment:
I would expect the test checking the right ephemeral post to be sent. Now you are not checking if the expected behavior is happening or not.